### PR TITLE
[#52] Include stack building instructions

### LINKED ISSUE #52: Add instructions to build project with stack
Currently, we have instructions to build project with `cabal`. But as per #42, we now support `stack` as well. So it would be great to have build instructions for both builds tools in README.

### PR DESCRIPTION
**Pull request by charrsky**
Resolves #52

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,9 +109,15 @@ will need to set it up with these steps:
...
 ```shell
 cabal build all
 cabal test --enable-tests --test-show-details=direct
 ```
+or `stack`:
+
+```shell
+stack build
+stack test
+```
```
Comment (chshersh) [inline README.md:123]: That's great! I'm thinking about making instructions more beginner-friendly so everyone without prior knowledge of Haskell tooling can have a quick start guide or at least pointers to where get the tooling. So something like this:

```markdown
To build the project and run the tests locally, you can use either `cabal` or `stack`.

### Cabal

1. Install [GHCup](...). It also install GHC (a Haskell compiler) and `cabal` (a Haskell build tool).
2. Build the project
3. Run the tests

### Stack

1. Install [Stack](...)
2. Build the project
3. Run the tests
```

**Review by chshersh — CHANGES_REQUESTED** [review]
Just one suggestion to improve on the structure and help other beginners even more 👍🏻

File: README.md [hunk 2]
```diff
@@ -109,9 +109,30 @@ will need to set it up with these steps:
...
 
-To build the project and run the tests, use `cabal`:
+## Building the project
+
+To build the project and run the tests locally, you can use either `cabal` or `stack`:
 
 ```shell
 cabal build all
-cabal test --enable-tests --test-show-details=direct
+cabal test --enable-test --test-show-details=direct
```
Comment (chshersh) [inline README.md:118]: Looks like a typo and it should be `--enabled-tests` and not `--enable-test` according the the `cabal test --help`:

```
$ cabal test --help
...
 --enable-tests                 Enable dependency checking and compilation for
                                test suites listed in the package description
                                file.
```

```suggestion
cabal test --enable-tests --test-show-details=direct
```

**Review by chshersh — COMMENTED** [review]
One minor comment but overall looks good 🙂

File: README.md (hunk 2 above)
Comment (charrsky) [reply README.md]: Yep, you are right. Done!

**Review by chshersh — APPROVED** [review]
Neat!

**Pull request merged**